code-challenges.test.js

```diff
@@ -21,8 +28,102 @@ const people = [
 ]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe("fullSentence", () => {
+  it("returns an array with a sentence about each person with their name capitalized", () => {
+    expect(fullSentence(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+//Test failed:  ReferenceError: fullSentence is not defined
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+
 // b) Create the function that makes the test pass.
 
+//Initial Pseudo:
+//Create a function called fullSentence.
+//Input: An array of objects with key-value pairs for a name and occupation.
+//Need to pull key-value pairs out of objects but keep data in arrays: possible method to do this
+//Need to combine arrays (name + occupation) together
+//Need to capitalize the first letter in the first and last name
+//Output: Returns an array with a sentence about each person with their name capitalized.
+
+//Final iteration of pseudocode inside function. 
+//I was super determined to use object deconstruction to make this function work. It was very difficult and took alot of time and reading documentation. I know you can simply use object dot notation to pull the key-pair values out but, wanted to explore this approach first. The fact that the output had to be one array with the string sentences at each element is what made this approach much more difficult. I knew there was a much more efficient way to do this problem, that refactored function is further below. 
+
+const fullSentence = (array) => {
+    //Empty array for object names
+    let newArray = []
+    //Empty array for object occupations
+    let newArray2 = []
+    //object deconstruction
+    const [element1, element2, element3] = array
```

Comment:
This setup is destructuring the array, which is great.  To destructure the objects, you can do this when you start iterating with a higher order function.  Ex: array.map(({name, occupation}) => name)